### PR TITLE
Adds test case for JSON anomaly logging

### PR DESCRIPTION
**Pull request by jlucovsky**
This changeset adds tests for JSON anomaly logging (part of [issue 2282](https://redmine.openinfosecfoundation.org/issues/2282)

### CODE REVIEW

File: tests/output-eve-anomaly/test.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+requires:
```
Comment (victorjulien) [inline tests/output-eve-anomaly/test.yaml]: requires should check for either: suri version 'min-version: 5.0.0' or the anomaly logger:
```
requires:
  features:
    - HAVE_LIBJANSSON
  files:
    - src/output-json-anomaly.c
```

File: tests/output-eve-anomaly/test.rules [hunk 2]
```diff
@@ -0,0 +1 @@
+alert tcp any 20 -> any any (msg:"Simple rule"; filestore;  sid:31; rev:1;)
```
Comment (victorjulien) [inline tests/output-eve-anomaly/test.rules]: no need to add a rules file if you're not checking the results or it matching anyway

File: tests/output-eve-anomaly/test.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  features:
+    - HAVE_LIBJANSSON
+
+checks:
+  - filter:
+      filename: eve.json
```
Comment (victorjulien) [inline tests/output-eve-anomaly/test.yaml]: no need to have this filename setting, eve.json is the default

File: tests/output-eve-anomaly/test.yaml (hunk 1 above)
Comment (victorjulien) [reply tests/output-eve-anomaly/test.yaml]: To clarrify: as-is the test will break on 4.1.x. It is supposed to be skipped on that version.

Comment (jlucovsky) [reply tests/output-eve-anomaly/test.yaml]: Will add the a `files:` section.

File: tests/output-eve-anomaly/test.rules (hunk 2 above)
Comment (jlucovsky) [reply tests/output-eve-anomaly/test.rules]: Will remove.

File: tests/output-eve-anomaly/test.yaml (hunk 3 above)
Comment (jlucovsky) [reply tests/output-eve-anomaly/test.yaml]: Will remove.

**Comment by jlucovsky** [comment]
Continued in #45

**Pull request closed without merging**